jormungandr-lib/src/interfaces/fragment_log_persistent.rs

```diff
@@ -14,3 +30,89 @@ pub struct PersistentFragmentLog {
     #[serde(with = "FragmentDef")]
     pub fragment: Fragment,
 }
+
+pub struct FileFragments {
+    reader: BufReader<fs::File>,
+    file_path: PathBuf,
+}
+
+pub struct FileFragmentsIterator {
+    reader: BufReader<fs::File>,
+    file_path: PathBuf,
+    counter: usize,
+}
+
+impl FileFragments {
+    pub fn from_path(file_path: PathBuf) -> std::io::Result<Self> {
+        fs::File::open(file_path.clone()).map(|file| Self {
+            reader: BufReader::new(file),
+            file_path,
+        })
+    }
+}
+
+impl IntoIterator for FileFragments {
+    type Item = Result<PersistentFragmentLog, Error>;
+    type IntoIter = FileFragmentsIterator;
+
+    fn into_iter(self) -> Self::IntoIter {
+        FileFragmentsIterator {
+            reader: self.reader,
+            file_path: self.file_path,
+            counter: 0,
+        }
+    }
+}
+
+impl Iterator for FileFragmentsIterator {
+    type Item = Result<PersistentFragmentLog, Error>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        // EOF reached when buffer is empty after reading successfully at least one.
+        // Then we stop the iterator.
+        if self.reader.buffer().is_empty() && self.counter != 0 {
```

Comment:
Wouldn't this break for an empty file?

Reply:
Not really break per se. But It would return an item with an Some(Err) because it couldn't deserialize.

Reply:
We can have empty files, so better to handle that case.

Reply:
I prefilled the buffer, so it is guaranteed to have something if files wasn't empty. If it is emtpy it will imideatly return a `None`, so it will be an empty iterator 😄